### PR TITLE
JRuby specs need to be able to pass args to in_sub_process

### PR DESCRIPTION
**Pull request by bf4**
Extracted from #134

### CODE REVIEW

File: lib/rspec/support/spec/in_sub_process.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module RSpec
   module Support
     module InSubProcess
-      if Process.respond_to?(:fork) && !(RUBY_PLATFORM == 'java' && RUBY_VERSION == '1.8.7')
+      if Process.respond_to?(:fork) && !(Ruby.jruby? && RUBY_VERSION == '1.8.7')
+
```
Comment (myronmarston) [inline lib/rspec/support/spec/in_sub_process.rb]: :+1:

File: lib/rspec/support/spec/in_sub_process.rb [hunk 2]
```diff
@@ -35,7 +36,7 @@ def in_sub_process(prevent_warnings=true)
           raise exception if exception
         end
       else
-        def in_sub_process
+        def in_sub_process(*)
```
Comment (myronmarston) [inline lib/rspec/support/spec/in_sub_process.rb]: :+1:

File: script/functions.sh [hunk 3]
```diff
@@ -112,7 +112,7 @@ function check_documentation_coverage {
 }
 
 function check_style_and_lint {
-  echo "bin/rubucop lib"
+  echo "bin/rubocop lib"
```
Comment (myronmarston) [inline script/functions.sh]: :+1:

File: spec/rspec/support/method_signature_verifier_spec.rb [hunk 4]
```diff
@@ -129,7 +129,7 @@ def arity_kw(x, y:1, z:2); end
               expect(valid?(nil, :a => 1)).to eq(false)
             end
 
-            it 'mentions the invalid keyword args in the error', :pending => RUBY_ENGINE == 'jruby' do
+            it 'mentions the invalid keyword args in the error' do
```
Comment (myronmarston) [inline spec/rspec/support/method_signature_verifier_spec.rb]: I suspect this change is only appropriate for jruby-head and not for stable released JRuby, right?  Were you planning on us merging with this?

**Comment by myronmarston** [comment]
The stuff I added :+1 to looks good to merge.  I think the other things are you trying to get the build passing against the current travis JRuby and we should remove that before merging?  If so, I'll be glad to merge this once that's been removed.

File: spec/rspec/support/method_signature_verifier_spec.rb (hunk 4 above)
Comment (bf4) [inline spec/rspec/support/method_signature_verifier_spec.rb]: It's up to you. It's more of a separation of concerns thing.

**Comment by bf4** [comment]
Ok, removing https://github.com/rspec/rspec-support/commit/a9a47661f29f7def3007463321a680d65e4b1c69 and https://github.com/rspec/rspec-support/commit/74e9bf6067d24077bda5bb4eb6a4fc804ce0150a

**Comment by myronmarston** [comment]
Cool, I'll merge when appveyor goes green.

**Comment by myronmarston** [comment]
Merged :).  Thanks!

**Pull request merged**